src/__tests__/render.js

```diff
@@ -87,3 +88,32 @@ test('unmounts', () => {
 
   expect(queryByTestId('node')).not.toBeInTheDocument()
 })
+
+test('use unmount before cleanup', () => {
+  const TestComponent = defineComponent((_, {slots}) => {
+    return () => slots.default?.()
+  })
+
+  const {getByTestId, unmount, queryByTestId} = render({
```

Comment:
I'd love to come up with the most succinct test that reproduces the issue. This would do it:

```js
test'unmounts when no wrapper element is present', () => {
  const Comp = defineComponent((_, ctx) => {
    return () => ctx.slots.default?.()
  })

  const {unmount} = render({
    render: () => h(Comp, () => h('div')),
  })

  unmount()
})
```